metpy/calc/basic.py

```diff
@@ -229,6 +229,57 @@ def heat_index(temperature, rh, mask_undefined=True):
     return hi
 
 
+@exporter.export
+@check_units(temperature='[temperature]', speed='[speed]')
+def feels_like(temperature, rh, speed, face_level_winds=False):
+    r"""Calculate the current feels like temperature.
+
+    Calculates the current feels like temperature based on the wind chill or heat index
+    as appropriate for the current conditions.
+
+    Parameters
+    ----------
+    temperature : `pint.Quantity`
+        The air temperature
+    rh : `pint.Quantity`
+        The relative humidity expressed as a unitless ratio in the range [0, 1].
+        Can also pass a percentage if proper units are attached.
+    speed : `pint.Quantity`
+        The wind speed at 10m.  If instead the winds are at face level,
+        `face_level_winds` should be set to `True` and the 1.5 multiplicative
+        correction will be applied automatically.
+    face_level_winds : bool, optional
+        A flag indicating whether the wind speeds were measured at facial
+        level instead of 10m, thus requiring a correction.  Defaults to
+        `False`.
+
+    Returns
+    -------
+    `pint.Quantity`
+        The corresponding feels like value(s)
+
+    See Also
+    --------
+    heat_index, windchill
+
+    """
+    wind_chill_temperature = windchill(temperature, speed, face_level_winds=face_level_winds,
+                                       mask_undefined=True)
+
+    heat_index_temperature = heat_index(temperature, rh, mask_undefined=True)
+
+    # Combine the heat index and wind chill arrays (no point has a value in both)
+    feels_like = np.ma.array(wind_chill_temperature.filled(1) *
+                             heat_index_temperature.filled(1),
+                             mask=(wind_chill_temperature.mask * heat_index_temperature.mask))
+
+    # Fill in missing areas where neither wind chill or heat index are applicable with the
+    # ambient temperature.
+    feels_like[feels_like.mask] = temperature[feels_like.mask]
+
+    return np.array(feels_like) * temperature.units
```

Comment:
Are you *positive* those are the correct units? It would seem safer to use one, or both, from the output of the calculations.

Reply:
What about:

```python
return np.array(feels_like.to(temperature.units)) * temperature.units
```

Reply:
Since this throws things directly into `np.ma.array`, I just feel like we need some kind of unit-support firewall to make sure things don't screw up. It also occurs to me that this should probably use `np.ma.where`:
```python
wc = windchill(...).to(temperature.units)
hi = heat_index(...).to(temperature.units)
feels_like = np.ma.where(wc.mask, hi, wc)
feels_like[feels_like.mask] = temperature[feels_like.mask]
return np.array(feels_like) * temperature.units
```
What do you think?